lib/Core/Traits/PagerfantaFindServiceSearchAdaptersTrait.php

```diff
@@ -0,0 +1,113 @@
+     * configured with
+     * ContentSearchAdapter and FindService
+     *
+     * @param Query $query
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function createContentSearchPager(Query $query, Request $request, $maxPerPage)
+    {
+        $adapter = new ContentSearchAdapter($query, $this->getSite()->getFindService());
+
+        return $this->getPagerfantaPager($adapter, $request, $maxPerPage);
+    }
+
+    /**
+     * Returns Pagerfanta pager that starts from first page
+     * and uses 'page' GET parameter from Request object
+     * configured with
+     * ContentSearchHitAdapter and FindService
+     *
+     * @param Query $query
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function createContentSearchHitPager(Query $query, Request $request, $maxPerPage)
+    {
+        $adapter = new ContentSearchHitAdapter($query, $this->getSite()->getFindService());
+
+        return $this->getPagerfantaPager($adapter, $request, $maxPerPage);
+    }
+
+    /**
+     * Returns Pagerfanta pager that starts from first page
+     * and uses 'page' GET parameter from Request object
+     * configured with
+     * LocationSearchAdapter and FindService
+     *
+     * @param LocationQuery $locationQuery
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function createLocationSearchPager(LocationQuery $locationQuery, Request $request, $maxPerPage)
+    {
+        $adapter = new LocationSearchAdapter($locationQuery, $this->getSite()->getFindService());
+
+        return $this->getPagerfantaPager($adapter, $request, $maxPerPage);
+    }
+
+    /**
+     * Returns Pagerfanta pager that starts from first page
+     * and uses 'page' GET parameter from Request object
+     * configured with
+     * LocationSearchHitAdapter and FindService
+     *
+     * @param LocationQuery $locationQuery
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function createLocationSearchHitPager(LocationQuery $locationQuery, Request $request, $maxPerPage)
+    {
+        $adapter = new LocationSearchHitAdapter($locationQuery, $this->getSite()->getFindService());
+
+        return $this->getPagerfantaPager($adapter, $request, $maxPerPage);
+    }
+
+    /**
+     * Shorthand method for creating Pagerfanta pager
+     * with preconfigured Adapter
+     *
+     * @param AdapterInterface $adapter
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function getPagerfantaPager(AdapterInterface $adapter, Request $request, $maxPerPage)
+    {
+        $pager = new Pagerfanta($adapter);
+        $pager->setMaxPerPage($maxPerPage);
+        $pager->setCurrentPage(
+            $request->query->get('page', 1)
```

Comment:
Why not transfer the current page with the method parameter to remove dependency on the `Request` object?

Reply:
But why ? On all projects we are getting `page` parameter from `Request` object, with this it would be automatically done for you.

Reply:
Because this way, I can't just simply use the trait to create paged output outside of controller.

What if I want to have two paged queries on the same page, one displaying a query in first page, one displaying a different query on second page?

Reply:
I that case you should avoid using this shorthand methods. The idea is to cover 80-90% of use cases with this.

Reply:
I disagree. The point of the trait is to be reusable in most usecases and using it outside of a controller where no request is readily available is a valid and often used usecase.

Moreover, using a page query parameter is only an implementation detail coming from Pagerfanta Bundle and not the Pagerfanta lib itself.

Reply:
I've updated code. wdyt @emodric ?

Reply:
Looks good :+1: